spec/index.md

```diff
@@ -844,6 +844,14 @@ It must be contained in the canister ranges of a subnet, otherwise the correspon
 
     -   Otherwise, the call is rejected by the system independently of the effective canister id.
 
+-   If the request is a query call to the Management Canister (`aaaaa-aa`), then:
+
+    -   If the call is to the `bitcoin_get_balance_query` or `bitcoin_get_utxos_query` method, then the effective canister id for this call must be the Management Canister (`aaaaa-aa`).
```

Comment:
@r-birkner Do the BN expect and enforce the condition on bitcoin query calls?

Reply:
Sorry for replying so late. I somehow missed it and Maksym had to remind me to reply.

Just to make sure: 
> If the request is a query call to the Management Canister (`aaaaa-aa`), then:
This means that the canister ID in the request body is set to `aaaaa-aa`?

As far as I can tell, we don't enforce that. The only thing we do is that if it is a query to the management canister for the two specified methods, we rewrite the canister ID to either the testnet or mainnet BTC canister depending on the network argument in the request.

I will double-check with the team to make sure my response is really correct :)

Reply:
> If the request is a query call to the Management Canister (aaaaa-aa), then:
> This means that the canister ID in the request body is set to aaaaa-aa?

Indeed, this means that the canister_id field in the CBOR map of the request's body is set to aaaaa-aa.

Reply:
It looks like (see [here](https://github.com/dfinity/ic/blob/d970aa69a9286c34d8cbf34985e3d0c44ca2e9bc/rs/execution_environment/src/query_handler.rs#L209)) the replica expects to see aaaaa-aa as the query's canister_id. So all the BN could do is to route the request to the bitcoin subnet and maybe also update the (effective) canister ID in the URL.

Reply:
> As far as I can tell, we don't enforce that.

I see. The replica doesn't enforce it either, but I guess I don't have a (security) argument to push for a change here.